File: the99thTime/Programs_v2/ImageResizeBenchmark.py
from PIL import Image
import os

def resize_images(inputFolder, outputFolder, newWidth, newHeight):
    inputFiles = [f for f in os.listdir(inputFolder) if f.lower().endswith(('.jpg', '.png', '.jpeg', '.bmp'))]

    for input_file in inputFiles:
        try:
            input_path = os.path.join(inputFolder, input_file)
            original_image = Image.open(input_path)

            resized_image = original_image.resize((newWidth, newHeight))

            output_file_name = "resized_" + input_file
            output_path = os.path.join(outputFolder, output_file_name)

            resized_image.save(output_path, "JPEG")

            print(f"Resized: {input_file}")
        except Exception as e:
            print(f"Error resizing: {input_file}")
            print(e)

if __name__ == "__main__":
    inputFolder = "Input" 
    outputFolder = "Output"
    newWidth = 400
    newHeight = 400

    if not os.path.exists(outputFolder):
        os.makedirs(outputFolder)

    resize_images(inputFolder, outputFolder, newWidth, newHeight)

File: the99thTime/Programs_v2/ImageResizeBenchmark.java
import java.awt.*;
import java.awt.image.BufferedImage;
import java.io.File;
import java.io.IOException;
import javax.imageio.ImageIO;

public class ImageResizeBenchmark {
    public static void main(String[] args) {
        String inputFolderPath = "Input";
        String outputFolderPath = "Output";
        int newWidth = 400;
        int newHeight = 400;

        File inputFolder = new File(inputFolderPath);
        File[] inputFiles = inputFolder.listFiles();

        if (inputFiles != null) {
            for (File inputFile : inputFiles) {
                try {
                    BufferedImage originalImage = ImageIO.read(inputFile);

                    BufferedImage resizedImage = new BufferedImage(newWidth, newHeight, BufferedImage.TYPE_INT_RGB);

                    Graphics2D g2d = resizedImage.createGraphics();
                    g2d.drawImage(originalImage, 0, 0, newWidth, newHeight, null);
                    g2d.dispose();

                    String outputFilePath = outputFolderPath + File.separator + "resized" + inputFile.getName();
                    File outputFile = new File(outputFilePath);
                    ImageIO.write(resizedImage, "jpg", outputFile);

                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }
        System.exit(1) ;
    }
}
